### ISSUE TITLE
Android only: setting `display: 'none'` style still renders visible borders

### ISSUE THREAD

**Issue by DSom20** [facebook/react-native#30338]
## Description
On Android only, an element with `display: 'none'` styling will still have its borders rendered.

## React Native version:
<details><summary>`npx react-native info` (using basic `expo init` managed workflow setup)</summary>
<p>

```
System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 60.95 MB / 8.00 GB
    Shell: 5.0.17 - /usr/local/bin/bash
  Binaries:
    Node: 14.10.0 - /usr/local/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: iOS 13.6, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6821437
    Xcode: 11.6/11E708 - /usr/bin/xcodebuild
  Languages:
    Java: 14.0.1 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: https://github.com/expo/react-native/archive/sdk-39.0.4.tar.gz => 0.63.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

</p>
</details>

## Steps To Reproduce

1. Provide both `display: 'none'` and any size border, eg `borderWidth: 10` to an element (eg a `View`)
  Eg, `<View style={{ display: 'none', borderWidth: 10}} />`
2. Sometimes to trigger the error I also have to wrap that view inside 2 containing views, the outer one having an `alignItems` style set, like so:
  ```javascript
      <View style={{ alignItems: 'center'}}>
        <View>
          <View style={{ borderWidth: 100, display: 'none'}} />
        </View>
      </View>
  ```

When testing on my Android emulator (Pixel_2_API_30), I don't need those wrapping views to trigger it. But when testing with an expo snack, it was necessary.

## Expected Results
No part of an element with `display: 'none'` styling should be rendered. Ie, even its borders should not be rendered.

## Snack, code example, screenshot, or link to a repository:
Interactive Snack [here](https://snack.expo.io/@dsom20/android-display-none-bug--interactive-example)
More minimal, static snack [here](https://snack.expo.io/@dsom20/android-display-none-bug--static-example)
Most minimal snack [here](https://snack.expo.io/@dsom20/android-display-none-bug--minimal-example)
   -Note: this last minimal snack is set to use Expo SDK 38. The error occurs using this very minimal code on the expo snack when using Expo SDK 38, but not when using SDK 39. However, when testing Expo SDK 39 on my Android emulator (Pixel_2_API_30), the error does occur. See [here](https://github.com/DSom20/react-native-android-display-none-bug) for the Github repo of this minimal example.

Screenshot examples:
- `View` with red border with default `display` style:
![display-flex](https://user-images.githubusercontent.com/48467559/98454639-ca7dae80-211b-11eb-954f-2f16920975af.png)
- `View` with red border with `display: 'none'`
![display-none](https://user-images.githubusercontent.com/48467559/98454642-cea9cc00-211b-11eb-860b-a1648ff2b8bd.png)

**Comment by finghtingjie** [comment]
i have the same problem, i finally solved it by return null instead of setting display: 'none'.

**Comment by DSom20** [comment]
@finghtingjie Ya, that's a good workaround for now. The workaround I'm using is to conditionally style the borders themselves as well. Ie, whenever I set `display: 'none'`, I also set `borderWidth: 0`. 

(I'm working with an API where I want to target only a single nested element, and the API doesn't seem to allow me to conditionally render that element itself like your method. I'm only given access to change its style.)

---

(Further detail regarding the API I'm working with if it matters for context: Specifically I'm working with [ReactNativeElements' Input](https://reactnativeelements.com/docs/input) component. I want a label to always be present, but conditionally render the InputComponent (and its container). I am able to conditionally pass in a `() => null` sort of thing as the InputComponent prop value. However the borders are on the container, and the container would still get rendered. The only prop for dealing with that container is `inputContainerStyle`. So that's where I end up passing in the conditional `display: 'none'` and `borderWidth: 0`.)
